libnavigation-base/src/main/java/com/mapbox/navigation/base/route/DirectionsSession.kt

```diff
@@ -5,13 +5,10 @@ import com.mapbox.navigation.base.route.model.Route
 
 interface DirectionsSession {
     fun getRoutes(): List<Route>
-    fun setOrigin(point: Point)
-    fun getOrigin(): Point
-    fun setWaypoints(points: List<Point>)
+    fun getOrigin(): Point?
     fun getWaypoints(): List<Point>
-    fun setDestination(point: Point)
-    fun getDestination(): Point
-    fun requestRoutes()
+    fun getDestination(): Point?
+    fun requestRoutes(origin: Point, waypoints: List<Point> = emptyList(), destination: Point)
```

Comment:
This is a public API and can be used from Java. We're using default parameters for `waypoints`. Are we missing `@JvmOverloads`? How are you envisioning using `requestRoutes` from Java?

Reply:
`@JvmOverloads` cannot be used with _interface_. When you use `Java` you should send `waypoints = emptyList()` explicitly. 

Please note that default params is sugar code, that could be used by kotlin's developers and sometimes make it easier for java's by `@JvmOverloads`